snapshot-labs/sx-monorepo
# feat: filters proposals by label

### Summary



Closes: https://github.com/snapshot-labs/workflow/issues/254
Depends on https://github.com/snapshot-labs/sx-monorepo/pull/914

This PR adds a new filter to the proposals page, to filter proposals by one or more labels.

Labels are filtered using an `OR` operator, meaning that if filtering by A and B, proposals with either A or B will be returned (and not proposals having both A and B)


![Screenshot 2024-10-21 at 17 16 35](https://github.com/user-attachments/assets/b8cc773e-cfc0-4f8a-8315-d13dfb19f5d1)

![Screenshot 2024-10-21 at 17 16 42](https://github.com/user-attachments/assets/93bbf0f4-1ab4-455d-915b-e2a3c2a0bb27)

![Screenshot 2024-10-21 at 17 16 49](https://github.com/user-attachments/assets/b030ee8a-d2a3-4e8b-aaf0-03e5a9da1567)

### How to test

1. Go to a space proposals page
2. Try to select some labels in the labels filter
3. It should refresh the proposal list, and show only proposals with the related labels

### Note

- Ability to link individual label to the proposals page will come in a later PR: https://github.com/snapshot-labs/sx-monorepo/pull/924
